packages/x-date-pickers/src/DatePicker/DatePicker.tsx

```diff
@@ -373,4 +373,12 @@ DatePicker.propTypes = {
   yearsPerRow: PropTypes.oneOf([3, 4]),
 } as any;
 
+/**
+ * Demos:
```

Comment:
Does it work in your IDE? 🤔 
In my case on VS Code it doesn't do squat... 🤷 
If I move this comment before the `const DatePicker =` then it does what we expect. 👍 
I feel that we should go that route and IMHO, maybe we could also move the `export` from the bottom to be in line, where the component is defined?
Or am I missing something that forces us to define it on the bottom..? 🤔 🙈

Reply:
I think it is because we do the `<Component>.proptypes = { ... }` and `<Component>.displayName = ...` before exporting. not sure if this would still work as before if we move the export up 🤷🏼

Reply:
I'm pretty sure that it would still work without issues. There are certain components following this approach, like the `DateCalendar`. 🤔

Reply:
You're right ... it's an immediate export though but should work the same across the components. I guess this is out of scope for this, so I'll create a separate ticket to check and unify the approach taken.

I guess we prefer to go with 
```
export const <Component> = ...
```

Right?

Reply:
> Does it work in your IDE? 🤔
Yeah, it did work ... maybe it caught the reference regardless

Reply:
> I guess we prefer to go with
> 
> ```
> export const <Component> = ...
> ```
> 
> Right?

AFAIK - yes, unless someone has different idea. 🤔 

> Yeah, it did work ... maybe it caught the reference regardless

Strange... And you are also on VS Code? 🤔

Reply:
> Strange... And you are also on VS Code? 🤔

Nope ... I am on Webstorm